Automattic/pocket-casts-android
# Track subscription data

Project https://github.com/Automattic/pocket-casts-android/issues/261

# Description

This tracks plus subscription data as predefined properties from the cached subscription.

## To test

1. Delete the app from your device.
2. Launch the app.
3. Since it is a fresh install and cached subscription is null, notice that analytics logs contain: `"plus_has_subscription":false`.
4. Gift plus plan to your account using web admin console and login with the account in the app. By now, cached subscription is updated in the app.
5. Background the app so that `application_closed` event gets triggered.
6. Notice that analytics logs contain plus subscription details : e.g. `"plus_has_subscription":true,"plus_has_lifetime":false,"plus_subscription_type":"plus","plus_subscription_platform":"gift","plus_subscription_frequency":"none"`.


# Checklist
N/A
